source/MRMesh/MRMeshLoadObj.cpp

```diff
@@ -138,113 +525,54 @@ tl::expected<std::vector<NamedMesh>, std::string> fromSceneObjFile( const char*
+            std::vector<MeshBuilder::VertDuplication> dups;
+            result.mesh = Mesh::fromTrianglesDuplicatingNonManifoldVertices(
+                VertCoords( points.begin() + minV, points.begin() + maxV + 1 ), t, &dups );
             t.clear();
-        }
-        currentObjName.clear();
-    };
 
-    Timer timer( "split by lines" );
-    std::vector<size_t> newlines{ 0 };
-    {
-        constexpr size_t blockSize = 4096;
-        const auto blockCount = (size_t)std::ceil( (float)size / blockSize );
-        constexpr size_t maxGroupCount = 256;
-        const auto blocksPerGroup = (size_t)std::ceil( (float)blockCount / maxGroupCount );
-        const auto groupSize = blockSize * blocksPerGroup;
-        const auto groupCount = (size_t)std::ceil( (float)size / groupSize );
-        assert( groupCount <= maxGroupCount );
-        assert( groupSize * groupCount >= size );
-        assert( groupSize * ( groupCount - 1 ) < size );
+            assert( uvCoords.size() >= maxV );
+            result.uvCoords = { begin( uvCoords ) + minV, begin( uvCoords ) + maxV + 1 };
+            result.uvCoords.resize( result.mesh.points.size() );
+            for ( const auto& dup : dups )
+                result.uvCoords[dup.dupVert] = result.uvCoords[dup.srcVert];
 
-        std::vector<std::vector<size_t>> groups( groupCount );
-        tbb::task_group taskGroup;
-        for ( size_t gi = 0; gi < groupCount; gi++ )
-        {
-            taskGroup.run( [&, i = gi]
+            VertHashMap dst2Src;
+            dst2Src.reserve( dups.size() );
+            for ( const auto& dup : dups )
+                dst2Src.emplace( dup.dupVert, dup.srcVert );
+
+            if ( mtl.has_value() && ! mtl->empty() )
             {
-                std::vector<size_t> group;
-                const auto begin = i * groupSize;
-                const auto end = std::min( ( i + 1 ) * groupSize, size );
-                for ( auto ci = begin; ci < end; ci++ )
-                    if ( data[ci] == '\n' )
-                        group.emplace_back( ci + 1 );
-                groups[i] = std::move( group );
-            } );
-        }
-        taskGroup.wait();
+                auto materialIt = mtl->find( currentMaterialName );
+                const auto mtlCopy = *mtl;
 
-        size_t sum = newlines.size();
-        std::vector<size_t> groupOffsets;
-        for ( const auto& group : groups )
-        {
-            groupOffsets.emplace_back( sum );
-            sum += group.size();
+                if ( materialIt == mtl->end() )
+                {
+                    materialIt = mtl->find( "default" );
+                }
+                if ( materialIt == mtl->end() )
+                {
+                    materialIt = mtl->begin();
+                }
+               
+                if ( !materialIt->second.diffuseTextureFile.empty() )
+                {
+                    result.pathToTexture = dir / materialIt->second.diffuseTextureFile;
```

Comment:
Can we avoid `dir / ` and just return relative path `materialIt->second.diffuseTextureFile`?